packages/grid/_modules_/grid/components/GridRow.tsx

```diff
@@ -33,7 +33,12 @@ export const GridRow = (props: GridRowProps) => {
   const publish = React.useCallback(
     (eventName: string) => (event: React.MouseEvent) => {
       // Ignore portal
-      if (!event.currentTarget.contains(event.target as HTMLElement)) {
+      // The target is not an element when triggered by a Select inside the cell
+      // See https://github.com/mui-org/material-ui/issues/10534
+      if (
+        (event.target as any).nodeType === 1 &&
+        !event.currentTarget.contains(event.target as Element)
```

Comment:
Could we wrap that into a function such as `isTargetWithinElement` or `isTargetFromSubTree` or ?
Make the code more readable

Reply:
This will trigger the discussion about when to create helpers or not. 😂 I would prefer to keep it as it is because it's only used in two places now. But if we start using more I will definitely create a helper for it.

Reply:
That logic is pretty complex so I would definitely abstract it. 
Normally you can use a rule of 3, here it's repeated twice already, + complexity and readability benefits => clean code

Reply:
If @oliviertassinari and @DanailH agree we can move this to a helper. I'm against it for the following reasons:
1. This conditional is only used in two places
2. We had past discussions about helpers and we didn't reach an agreement
3. 100% times I see a helper I go to its definition to see what it does, because I don't trust the name
4. It makes the code cleaner but the developer has to navigate to more files to understand 
5. By the name of the helper I don't know if `currentTarget` or `target` is used.

Reply:
In that particular case, I'm more in favour of keeping it like it is. If you name the helper `isTargetWithinElement` I would still need to check its implementation to understand what it does. I'm ok with helpers like `isSpaceKey` for example as it is clear what the helper does.

Reply:
I'm all 💯 for https://github.com/mui-org/material-ui-x/pull/1772/files#r649207549. Some of the arguments are biased, some are not, but the arguments resonate with me. I also **never** trust a module I don't know by heart (nor I think anyone should ever), having to check the content of small helpers is mentally draining. Two repetitions are not a pattern.